From 7ab71279ec8a470f53c2da2a5e44b1e498a3aebb Mon Sep 17 00:00:00 2001
From: Sam Hartman <hartmans@debian.org>
Date: Mon, 26 Dec 2011 18:05:13 -0500
Subject: =?UTF-8?q?ticket:=20new=0Asubject:=20fix=20ksu=20environment=20vari?=
 =?UTF-8?q?able=20handling?=

Fix error messages from ksu

Patch-Category: to-submit
Patch-Name: ksu_environment_fix
---
 src/clients/ksu/ksu.h  |    8 ++++++--
 src/clients/ksu/main.c |    2 +-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/clients/ksu/ksu.h b/src/clients/ksu/ksu.h
index f2c0811..cbc4a57 100644
--- a/src/clients/ksu/ksu.h
+++ b/src/clients/ksu/ksu.h
@@ -55,8 +55,12 @@
 
 #define CHUNK 3
 #define CACHE_MODE 0600
-#define MAX_CMD 2048 /* this is temp, should use realloc instead,
-                        as done in most of the code */
+#define MAX_CMD 2048 /* this is temp, should use realloc instead,          
+			as done in most of the code */       
+
+#ifndef MAXPATHLEN
+# define MAXPATHLEN 4096
+#endif		      
 
 
 extern int optind;
diff --git a/src/clients/ksu/main.c b/src/clients/ksu/main.c
index d9e40bb..233eb52 100644
--- a/src/clients/ksu/main.c
+++ b/src/clients/ksu/main.c
@@ -697,7 +697,7 @@ main (argc, argv)
     }
 
     if(set_env_var( "HOME", target_pwd->pw_dir)){
-        fprintf(stderr, _("ksu: couldn't set environment variable USER\n"));
+        fprintf(stderr, _("ksu: couldn't set environment variable HOME\n"));
         sweep_up(ksu_context, cc_target);
         exit(1);
     }
